File: apps/sdkwork-appstore-h5/packages/sdkwork-appstore-h5-ai-hub/src/services/aiHubService.ts
import type { AppstoreAppSdkClient } from '@sdkwork/appstore-h5-core';

import type { AiHubPageResult } from '../types/aiHubModels';

/**
 * Ai Hub service.
 *
 * The generated app SDK clients are injected by the application root bootstrap
 * (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
 * never issues raw HTTP.
 */
export class AiHubService {
  readonly capability = 'ai-hub';

  constructor(private readonly client: AppstoreAppSdkClient) {}

  empty(): AiHubPageResult<never> {
    return { items: [] };
  }
}

File: apps/sdkwork-appstore-flutter-mobile/packages/sdkwork_appstore_flutter_mobile_ai_hub/lib/src/services/ai_hub_service.dart
import 'package:sdkwork_appstore_flutter_mobile_core/sdkwork_appstore_flutter_mobile_core.dart';

import '../models/ai_hub_models.dart';

/// Ai Hub service.
///
/// The App Store app SDK clients are injected by the root bootstrap
/// (`APP_SDK_INTEGRATION_SPEC.md`); this package never constructs a client and
/// never issues raw HTTP.
class AiHubService {
  const AiHubService({required this.clients});

  final AppstoreAppSdkClients clients;

  String get capability => 'ai-hub';

  AiHubPageResult<Never> empty() =>
      const AiHubPageResult<Never>(items: <Never>[]);
}
